### PR TITLE
Byte data type

### PR DESCRIPTION
**Pull request by R4PaSs**
As discussed in #1267, we need a Byte data type to manipulate low-level data as numerics instead of how it is done for now using chars.

This PR will serve as base for future updates, notably related to #1370 which will introduce a way to express Bytes in their literal form.

### CODE REVIEW

**Comment by BlackMinou** [comment]
:shipit: 
black-minou-mini

**Comment by xymus** [comment]
+1

In the next PR, please add a test with Byte in test_ffi_c_types.nit to make sure it works with the FFI.

File: lib/standard/kernel.nit [hunk 1]
```diff
@@ -402,6 +409,107 @@ universal Float
...
+	redef fun !=(i) is intern
+	redef fun output is intern
+
+	redef fun <=(i) is intern
+	redef fun <(i) is intern
+	redef fun >=(i) is intern
+	redef fun >(i) is intern
+	redef fun +(i) is intern
+
+	# On an Byte, unary minus will return `255 - self`
```
Comment (privat) [inline lib/standard/kernel.nit]: (256 - self) % 256

-0 == 0
-1 == 255

**Comment by privat** [comment]
test this please

**Comment by jpages** [comment]
+1

**Comment by R4PaSs** [comment]
Reroll: Fixed documentation of Byte::to_s and changed semantics of Byte::unary_minus for compliance with the C standard

**Comment by R4PaSs** [comment]
test this please

**Comment by R4PaSs** [comment]
m67vr

**Comment by R4PaSs** [comment]
Reroll post-#1417, tests are now passing

**Pull request merged**